libs/SalesforceSDKCore/SalesforceSDKCore/Classes/KeyValueEncryptedFileStore.swift

```diff
@@ -0,0 +1,395 @@
+            return false
+        }
+        guard let fileURL = encodedUrl(forKey: key, function: #function) else {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to construct file URL")
+            return false
+        }
+
+        do {
+            try encryptedData.write(to: fileURL)
+        } catch {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Error writing data to file: \(error)")
+            return false
+        }
+
+        return true
+    }
+
+    /// Accesses the value associated with the given key for reading and writing.
+    @objc public subscript(key: String) -> String? {
+        get {
+            guard let fileURL = encodedUrl(forKey: key, function: #function) else {
+                SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to construct file URL")
+                return nil
+            }
+            guard FileManager.default.fileExists(atPath: fileURL.path) else {
+                return nil
+            }
+
+            do {
+                let encryptedData = try Data(contentsOf: fileURL)
+                guard let decryptedData = encryptionKey.decryptData(encryptedData) else {
+                    SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to decrypt file at path '\(fileURL.path)'")
+                    return nil
+                }
+                return String(data: decryptedData, encoding: .utf8)
+            } catch {
+                SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Error reading file at path '\(fileURL.path)': \(error)")
+                return nil
+            }
+        }
+
+        set {
+            if let newValue = newValue {
+                saveValue(newValue, forKey: key)
+            } else {
+                removeValue(forKey: key)
+            }
+        }
+    }
+    /// Removes entry for given key.
+    /// - Parameter key: The key associated with the entry to remove.
+    /// - Returns: True if the entry is successfully removed or doesn't exist, false otherwise.
+    @objc @discardableResult public func removeValue(forKey key: String) -> Bool {
+        guard let fileURL = encodedUrl(forKey: key, function: #function) else {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Unable to construct file URL")
+            return false
+        }
+        guard FileManager.default.fileExists(atPath: fileURL.path) else {
+            return true
+        }
+
+        do {
+            try FileManager.default.removeItem(at: fileURL)
+            return true
+        } catch {
+            SFSDKCoreLogger.e(KeyValueEncryptedFileStore.self, message: "\(#function): Error removing file at path '\(fileURL.path)': \(error)")
+            return false
+        }
+    }
+
+    /// Removes all contents of the store.
+    @objc public func removeAll() {
```

Comment:
We should call this when the user logs out, to ensure that all data is wiped when logout occurs.